No jira fix bug in move to top level singleton

jebentier opened the pull request:
## [0.10.5] - Unreleased
### Fixed
- Fixed bug where setting a monitor instance at the `ProcessSettings` and the `ProcessSettings::Monitor`
can cause unexpected errors due to two monitors being configured.

nburwell commented on spec/lib/process_settings_spec.rb:
```diff
@@ -12,7 +12,6 @@
 
     describe 'when lazy loading' do
       before do
-        expect(ActiveSupport::Deprecation).to receive(:warn).with("lazy creation of Monitor instance is deprecated and will be removed from ProcessSettings 1.0")
```
Shouldn't this test still warn because `instance` will call `default_instance`, where the warning is now.

ColinDKelley replied:
Agreed!

jebentier replied:
I've gone ahead and added the expectation back.  This test is now showing that we are going to get a deprecation warning from the lazy loading of `Monitor.instance` which already calls out to switch to setting the monitor instance on `ProcessSettings`.

ColinDKelley commented:
I think we still want to deprecate anyone who uses the lazy create of the instance. I laid out an approach here:
https://invoca.slack.com/archives/CRGB17J59/p1590022279244700
There are other ways to do it, but I think we need _something_.

jebentier commented:
@ColinDKelley because `default_instance` is already deprecated within `ProcessSettings::Monitor` we're still getting a deprecation warning, I just removed the duplicate testing of this logic from the test.  I've added it back now, to show in the tests that calling `ProcessSettings.instance` to lazy load the `default_instance` from `ProcessSettings::Monitor` will give an appropriate deprecation warning telling the developer to start using the top level `ProcessSettings.instance` instead.

ColinDKelley commented on lib/process_settings/monitor.rb:
```diff
@@ -31,11 +31,11 @@ def clear_instance
       end
 
       def default_instance
+        ActiveSupport::Deprecation.warn("ProcessSettings::Monitor.default_instance is deprecated and will be removed in v1.0. Use ProcessSettings.instance instead.")
```
```suggestion
        ActiveSupport::Deprecation.warn("ProcessSettings::Monitor.default_instance is deprecated and will be removed in v1.0. Assign a `FileMonitor` object to `ProcessSettings.instance=` instead.")
```

jebentier replied:
I've gone ahead and applied this change. I think this resolves what we're both trying to do, which is mainly the following:
1. Throw as few deprecation warnings from a single path as possible
2. Make sure that what ever deprecation warnings are raised, they are accurate and actionable.
